pkg/virt-controller/services/template.go

```diff
@@ -1502,7 +1502,9 @@ func (t *TemplateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		resourceRules: []VMIResourceRule{
 			// Run overcommit first to avoid overcommitting overhead memory
 			NewVMIResourceRule(func(vmi *v1.VirtualMachineInstance) bool {
-				return t.clusterConfig.GetMemoryOvercommit() != 100
+				return !hasMemoryRequest(vmi) &&
+					hasGuestMemoryOrHugePages(vmi) &&
```

Comment:
Do you have any e2e coverage for this? I thought the mutation webhook always used the guest memory value as an initial resource request value if one wasn't provided:

https://github.com/kubevirt/kubevirt/blob/20a24d59aa9bb608b58c57ee15d8b80bb3e7d1a2/pkg/defaults/defaults.go#L279-L296

Reply:
Okay I think I understand what the problem is. `./examples/vmi-fedora.yaml` does not overcommit either (without limits), since `!hasMemoryRequest(vmi)` is always false.

As stated, this approach needs to be re-thinked.